src/lib/components/multiselect/MultiSelect.component.js

```diff
@@ -0,0 +1,420 @@
+        label: newOption['label'].value, 
+        content: newOption['title'].value
+      });
+            
+      newOption['title'].value = '';
+      newOption['label'].value = '';
+
+      return {
+        ...state,
+        options,
+        newOption,
+        currentInput: 'title'
+      };
+    }
+
+    return state;
+  }
+
+  switch (action.type) {
+    case 'change': return changeOption(action);
+    case 'check': return checkOption(action);
+    case 'remove': return removeOption(action);
+    case 'add': return addOption();
+    default: return state;
+  }
+}
+
+function MultiSelectOption(props: MultiSelectOptionProps) {
+  const { checked, label, content, onClick, onCheck } = props;
+
+  return (
+    <MultiSelectOptionContainer>
+      <MultiSelectContent>
+        <h4 className="content-title">{content}</h4>
+        <h4 className="content-label">{label}</h4>
+        <button className="remove-button" onClick={onClick}>
+          <RemoveIcon />
+        </button>
+      </MultiSelectContent>
+      <MultiSelectMark>
+        <input type="checkbox" className="checkbox" onChange={onCheck} checked={checked} />
+      </MultiSelectMark>
+    </MultiSelectOptionContainer>
+  )
+}
+
+function MultiSelectForm(props: MultiSelectFormProps) {
+  const { value, placeholder, onChange, onClick } = props;
+
+  return (
+    <MultiSelectFormContainer>
+      <input
+        className='input'
+        value={value}
+        placeholder={placeholder}
+        onChange={onChange}
+      />
+      <button 
+        type='button'
+        onClick={onClick}
+        className='add-button'
+      >
+        <AddIcon />
+      </button>
+    </MultiSelectFormContainer>
+  )
+}
+
+function MultiSelect(props: MultiSelectProps) {
+  const [state, dispatch] = useReducer(reducer, props);
+
+  return (
+    <MultiSelectContainer
+      width={state.width}
+    >
+      <BlockContainer>
+        <h3 className="title-multi-select">{state.title}</h3>
+        <h3 className="label-checkbox-multi-select">{state.labelCheckBox}</h3>
+      </BlockContainer>
+      <BlockContainer>
+        {
+          state.options.map(option =>
+            <MultiSelectOption 
+              checked={option.checked}
+              label={option.label}
+              content={option.content}
+              key={option.content}
+              onClick={() => dispatch({ type: 'remove', option })}
+              onCheck={() => dispatch({ type: 'check', option })}
+            />  
+          )
+        }
+      </BlockContainer>
+      <BlockContainer>
+        <MultiSelectForm 
+          value={state.newOption[state.currentInput].value}
+          placeholder={state.newOption[state.currentInput].placeholder}
+          onChange={e => dispatch({ type: 'change', newValue: e.target.value })}
+          onClick={() => dispatch({ type: 'add' })}
+        />
+      </BlockContainer>
+    </MultiSelectContainer>
+  );
+}
+
+MultiSelect.defaultProps = defaultProps;
```

Comment:
You should remove defaultProps since it will be deprecated soon. You can use `Default Function arguments` instead